modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java

```diff
@@ -140,15 +141,25 @@ public ConfigurationInstance get(Event event) {
         if (connectionProviderResolver.getResolverSet().isPresent()) {
           providerResult = ((ResolverSet) connectionProviderResolver.getResolverSet().get()).resolve(resolvingContext);
         }
-        return getConfiguration(new Pair<>(result, providerResult), (CoreEvent) event);
+
+        if (muleContext.isStopping() || muleContext.isStopped()) {
```

Comment:
this if seems to be inverted, but also conceptually wrong.. 

if the context is not stopping, then you need to do getConfiguration() as usual so that the cache is used. If the context IS stopping, then no new config should be created. Also, the compute method you added should never be explicitly called. If a dynamic config is not added to the cache, then it will never be stopped and will generate a leak.

Reply:
the cache is a loading cache, that's the issue, doing a get on it populates it, that's what i'm avoiding with this change, and generating a different compute method

Reply:
I know.. my point is that a get() to a dynamic config should simply fail when the context is stopping.